pypeit/spectrographs/spectrograph.py

```diff
@@ -627,6 +629,25 @@ def list_detectors(self, mosaic=False):
         dets = self.allowed_mosaics if mosaic else range(1,self.ndet+1)
         return np.array([self.get_det_name(det) for det in dets])
 
+    def parse_raw_files(self, fitstbl, det=1, ftype=None):
+        """
+        Parse the list of raw files with given frame type and detector.
+
+        Args:
+            fitstbl (`astropy.table.Table`_):
+                Table with metadata of the raw files to parse.
+            det (:obj:`int`, optional):
+                1-indexed detector number to parse.
+            ftype (:obj:`str`, optional):
+                Frame type to parse. If None, no frames are parsed
+                and the indices of all frames are returned.
+
+        Returns:
+            `numpy.ndarray`_: The indices of the raw files in the fitstbl that are parsed.
+
+        """
+
+        return np.arange(len(fitstbl))
```

Comment:
I'm not sure I understand the general application of this function.  Given that it's only used for slitless flats right now, do we need a general function?

Reply:
This is actually spectrograph dependent, this is why it's here. It is currently only used for `keck_hires`, but I tried to make it generic enough so that it could be used for other purposes (still spectrograph-specific). I have added more in the docstring.